### PR TITLE
Update instructions of authentication portal external hosting

### PR DESCRIPTION
**Pull request by AnuradhaSK**
## Purpose
Fix https://github.com/wso2/product-is/issues/12984
and update jar files as per the working sample https://github.com/wso2/samples-is/blob/master/host-endpoints-externally/setup-authentication-endpoint.sh

### CODE REVIEW

File: en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md [hunk 1]
```diff
@@ -83,28 +91,48 @@ Server).
...
+
+5. Copy the following .jar files from the `<IS_HOME>/repository/components/tools/forget-me/lib/` directory to the `<WEBAPP_HOME>/authenticationendpoint/WEB-INF/lib` directory. 
 
-2.  Copy the following .jar files from the `IS_HOME>/lib/runtimes/cxf/` directory to the `<WEBAPP_HOME>/authenticationendpoint/WEB-INF/lib`
-    directory.  
+    ```xml
+    - log4j-*.jar
+    - pax-logging-api-*.jar
+    ```
+6. Copy the .jar files from `<IS_HOME>repository/components/features/org.wso2.carbon.identity.application.authentication.framework.server_*/runtimes/cxf3/org.wso2.carbon.identity.application.authentication.endpoint.util-*.jar` to the`<WEBAPP_HOME>/authenticationendpoint/WEB-INF/lib` directory.
```
Comment (nilminiwso2) [inline en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md:118]:
```suggestion
6. Copy `<IS_HOME>repository/components/features/org.wso2.carbon.identity.application.authentication.framework.server_*/runtimes/cxf3/org.wso2.carbon.identity.application.authentication.endpoint.util-*.jar` to the`<WEBAPP_HOME>/authenticationendpoint/WEB-INF/lib` directory. 
```

correct?

File: en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md [hunk 2]
```diff
@@ -83,28 +91,48 @@ Server).
...
-    directory.  
+    ```xml
+    - log4j-*.jar
+    - pax-logging-api-*.jar
+    ```
+6. Copy the .jar files from `<IS_HOME>repository/components/features/org.wso2.carbon.identity.application.authentication.framework.server_*/runtimes/cxf3/org.wso2.carbon.identity.application.authentication.endpoint.util-*.jar` to the`<WEBAPP_HOME>/authenticationendpoint/WEB-INF/lib` directory. 
     
+7. Copy the `RecoveryEndpointConfig.properties` file from the `<IS-HOME>/repository/conf/identity` directory to the `<WEBAPP_HOME>/authenticationendpoint/WEB-INF/classes` directory.
+
+8. Navigate to `<WEBAPP_HOME>/authenticationendpoint/WEB-INF/classes/RecoveryEndpointConfig.properties` file and uncomment the following line.
```
Comment (nilminiwso2) [inline en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md:122]:
```suggestion
8. Open the `<WEBAPP_HOME>/authenticationendpoint/WEB-INF/classes/RecoveryEndpointConfig.properties` file and uncomment the following line.
```

File: en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md [hunk 3]
```diff
@@ -83,28 +91,48 @@ Server).
...
     ```xml
-    - javax.ws.rs-
-    - api `-2.0-m10.jar
-    - cxf `-bundle-2.7.16.wso2v1.jar
-    - neethi `-3.0.3.jar
-    - wsdl4j-1.6.3.jar
+    identity.server.service.contextURL=https://localhost:9443
+    ```
+9. Navigate to `<WEBAPP_HOME>/authenticationendpoint/WEB-INF/classes/EndpointConfig.properties` file and change the following line,
```
Comment (nilminiwso2) [inline en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md:126]:
```suggestion
9. Open the `<WEBAPP_HOME>/authenticationendpoint/WEB-INF/classes/EndpointConfig.properties` file and change the following line,
```

File: en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md [hunk 4]
```diff
@@ -83,28 +91,48 @@ Server).
...
+    ```
+    identity.server.origin=\${carbon.protocol}://\${carbon.host}:\${carbon.management.port}
+    ```
+    as follows and update the Keystore file paths on the same file accordingly.
```
Comment (nilminiwso2) [inline en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md:130]:
```suggestion
    as follows and update the Keystore file paths on the same file accordingly.
```

File: en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md [hunk 5]
```diff
@@ -83,28 +91,48 @@ Server).
...
     ```xml
-    - javax.ws.rs-
-    - api `-2.0-m10.jar
-    - cxf `-bundle-2.7.16.wso2v1.jar
-    - neethi `-3.0.3.jar
-    - wsdl4j-1.6.3.jar
+    identity.server.service.contextURL=https://localhost:9443
+    ```
+9. Navigate to `<WEBAPP_HOME>/authenticationendpoint/WEB-INF/classes/EndpointConfig.properties` file and change the following line,
+    ```
```
Comment (nilminiwso2) [inline en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md:127]: Don't need this code block here if we are using the above line.

File: en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md [hunk 6]
```diff
@@ -83,28 +91,48 @@ Server).
...
+    identity.server.origin=\${carbon.protocol}://\${carbon.host}:\${carbon.management.port}
+    ```
+    as follows and update the Keystore file paths on the same file accordingly.
+    ```
+    identity.server.origin=https://localhost:9443
     ```
 
-3.  Uncomment following section in `<WEBAPP_HOME>/authenticationendpoint/WEB-INF/web.xml` and point to identity server URLs.
+10. Uncomment following section in `<WEBAPP_HOME>/authenticationendpoint/WEB-INF/web.xml` and point to identity server URLs.
```
Comment (nilminiwso2) [inline en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md:135]:
```suggestion
10. Uncomment the following section in the `<WEBAPP_HOME>/authenticationendpoint/WEB-INF/web.xml` file and point to the identity server URLs.
```

File: en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md [hunk 7]
```diff
@@ -124,7 +152,7 @@ Server).
     ...
     ```
 
-4.  Add the following configurations to the `<IS_HOME>/repository/conf/deployment.toml` file.
+11. Add the following configurations to the `<IS_HOME>/repository/conf/deployment.toml` file.
```
Comment (nilminiwso2) [inline en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md:155]:
```suggestion
11. Add the following configurations to the `<IS_HOME>/repository/conf/deployment.toml` file:
```

File: en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md [hunk 8]
```diff
@@ -140,7 +168,7 @@ Server).
     request_missing_claims_url = "https://localhost.com:8443//authenticationendpoint/claims.do"
     ```
 
-5.  Add the following configurations to the `<IS_HOME>/repository/conf/deployment.toml` file to point to the authentication endpoint hosted outside the WSO2 server.
+12. Add the following configurations to the `<IS_HOME>/repository/conf/deployment.toml` file to point to the authentication endpoint hosted outside the WSO2 server.
```
Comment (nilminiwso2) [inline en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md:171]:
```suggestion
12. To point to the authentication endpoint hosted outside the WSO2 server, add the following configurations to the `<IS_HOME>/repository/conf/deployment.toml` file.
```

File: en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md [hunk 9]
```diff
@@ -158,7 +186,7 @@ Server).
     retry= "https://localhost:8443/authenticationendpoint/retry.do"
     ```
 
-6.  Import the public certificate of the identity server to the javaca certs (or web-serverstruststore) of the JVM that the authenticationendpoint is running.
+13. Import the public certificate of the identity server to the javaca certs (or web-serverstruststore) of the JVM that the authenticationendpoint is running.
```
Comment (nilminiwso2) [inline en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md:189]:
```suggestion
13. Import the public certificate of the identity server to the `javaca` certs (or `web-serverstruststore`) of the JVM where the authentication endpoint is running.
```

Comment (nilminiwso2) [reply en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md:189]: What is `javaca` certs? Is this supposed to be a truststore?

File: en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md [hunk 10]
```diff
@@ -168,7 +196,7 @@ Server).
     keytool -import -alias wso2carbon -keystore  $WEB_APP_TRUSTSTORE -file wso2carbon.cer
     ```
 
-7.  Import the public certificate of the Web\_server’s keystore to the Identity Server truststore.
+14. Import the public certificate of the Web\_server’s keystore to the Identity Server truststore.
```
Comment (nilminiwso2) [inline en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md:199]:
```suggestion
14. Import the public certificate of the web server’s keystore to the Identity Server truststore.
```

File: en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md [hunk 11]
```diff
@@ -365,7 +393,19 @@ This is an additional improvement which enables hosting `accountrecoveryendpoin
     …...
     ```
 
-8.  Add the following configurations to the `<IS_HOME>/repository/conf/deployment.toml` file.
+8.  Navigate to `<TOMCAT_HOME>/authenticationendpoint/WEB-INF/classes/RecoveryEndpointConfig.properties` file and uncomment the following line.
```
Comment (nilminiwso2) [inline en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md:396]:
```suggestion
8.  Open the `<TOMCAT_HOME>/authenticationendpoint/WEB-INF/classes/RecoveryEndpointConfig.properties` file and uncomment the following line:
```

File: en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md [hunk 12]
```diff
@@ -365,7 +393,19 @@ This is an additional improvement which enables hosting `accountrecoveryendpoin
...
+    ```
+    identity.server.service.contextURL=https://localhost:9443
+    ```
+9.  Navigate to `<TOMCAT_HOME>/authenticationendpoint/WEB-INF/classes/EndpointConfig.properties` file and change the following line,
```
Comment (nilminiwso2) [inline en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md:400]:
```suggestion
9.  Open the `<TOMCAT_HOME>/authenticationendpoint/WEB-INF/classes/EndpointConfig.properties` file and change the following line:
```

File: en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md [hunk 13]
```diff
@@ -365,7 +393,19 @@ This is an additional improvement which enables hosting `accountrecoveryendpoin
...
+    ```
+    identity.server.origin=\${carbon.protocol}://\${carbon.host}:\${carbon.management.port}
+    ```
+    as follows and update the Keystore file paths on the same file accordingly.
+    ```
+    identity.server.origin=https://localhost:9443
+    ```
```
Comment (nilminiwso2) [inline en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md:407]: The following is a new step yeah?

```
10. Update the keystore file paths on the same file accordingly.
```

File: en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md [hunk 14]
```diff
@@ -365,7 +393,19 @@ This is an additional improvement which enables hosting `accountrecoveryendpoin
...
-8.  Add the following configurations to the `<IS_HOME>/repository/conf/deployment.toml` file.
+8.  Navigate to `<TOMCAT_HOME>/authenticationendpoint/WEB-INF/classes/RecoveryEndpointConfig.properties` file and uncomment the following line.
+    ```
+    identity.server.service.contextURL=https://localhost:9443
+    ```
+9.  Navigate to `<TOMCAT_HOME>/authenticationendpoint/WEB-INF/classes/EndpointConfig.properties` file and change the following line,
+    ```
+    identity.server.origin=\${carbon.protocol}://\${carbon.host}:\${carbon.management.port}
+    ```
+    as follows and update the Keystore file paths on the same file accordingly.
```
Comment (nilminiwso2) [inline en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md:404]:
```suggestion
```

File: en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md [hunk 15]
```diff
@@ -365,7 +393,19 @@ This is an additional improvement which enables hosting `accountrecoveryendpoin
...
+    ```
+    identity.server.origin=https://localhost:9443
+    ```
+10. Add the following configurations to the `<IS_HOME>/repository/conf/deployment.toml` file.
```
Comment (nilminiwso2) [inline en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md:408]:
```suggestion
10. Add the following configurations to the `<IS_HOME>/repository/conf/deployment.toml` file:
```

File: en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md [hunk 16]
```diff
@@ -374,7 +414,7 @@ This is an additional improvement which enables hosting `accountrecoveryendpoin
     request_missing_claims_url="https://localhost:8443/authenticationendpoint/claims.do"
     ```
 
-9.  Add the following configurations to the `<IS_HOME>/repository/conf/deployment.toml` file to point to the authentication endpoint hosted outside the WSO2 server.
+11. Add the following configurations to the `<IS_HOME>/repository/conf/deployment.toml` file to point to the authentication endpoint hosted outside the WSO2 server.
```
Comment (nilminiwso2) [inline en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md:417]:
```suggestion
11. To point to the authentication endpoint hosted outside the WSO2 server, add the following configurations to the `<IS_HOME>/repository/conf/deployment.toml` file.
```

File: en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md [hunk 17]
```diff
@@ -392,14 +432,14 @@ This is an additional improvement which enables hosting `accountrecoveryendpoin
...
     ```
 
-10. Start both Identity Server and tomcat and access `https://localhost:9443/dashboard`. Now you can see that the authentication is redirected to: `https://localhost:8443/authenticationendpoint/login.do`
+12. Start both Identity Server and tomcat and access `https://localhost:9443/dashboard`. Now you can see that the authentication is redirected to: `https://localhost:8443/authenticationendpoint/login.do`
 
     Now let’s take out account recovery endpoint into the external
     Tomcat server as well.
 
-11. Run `setup-accountrecovery.sh` obtained from [step 2](#HostingAuthenticationEndpointonaDifferentServer-step2) and follow the instructions.
+13. Run `setup-accountrecovery.sh` obtained from [step 2](#HostingAuthenticationEndpointonaDifferentServer-step2) and follow the instructions.
```
Comment (nilminiwso2) [inline en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md:440]:
```suggestion
13. Run the `setup-accountrecovery.sh` script, which you obtained from [step 2](#HostingAuthenticationEndpointonaDifferentServer-step2) and follow the instructions.
```

File: en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md [hunk 18]
```diff
@@ -392,14 +432,14 @@ This is an additional improvement which enables hosting `accountrecoveryendpoin
...
 
-12. Change the following section in `<TOMCAT_HOME>/webapps/authenticationendpoint/WEB-INF/web.xml` file and point to `IdentityManagementEndpointContextURL` into tomcat URL.
+14. Change the following section in `<TOMCAT_HOME>/webapps/authenticationendpoint/WEB-INF/web.xml` file and point to `IdentityManagementEndpointContextURL` into tomcat URL.
```
Comment (nilminiwso2) [inline en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md:442]:
```suggestion
14. Change the following section in the `<TOMCAT_HOME>/webapps/authenticationendpoint/WEB-INF/web.xml` file and point `IdentityManagementEndpointContextURL` to the Tomcat URL.
```

File: en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md [hunk 19]
```diff
@@ -410,7 +450,7 @@ This is an additional improvement which enables hosting `accountrecoveryendpoin
     …
     ```
 
-13. Uncomment and change the user portal reference in the `<TOMCAT_HOME>/accountrecoveryendpoint/WEB-INF/web.xml` file.
+15. Uncomment and change the user portal reference in the `<TOMCAT_HOME>/accountrecoveryendpoint/WEB-INF/web.xml` file.
```
Comment (nilminiwso2) [inline en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md:453]:
```suggestion
15.  Open the `<TOMCAT_HOME>/accountrecoveryendpoint/WEB-INF/web.xml` file, uncomment the section given below, and update the user portal reference.
```

File: en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md (hunk 9 above)
Comment (AnuradhaSK) [reply en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md:189]: javaca is where Java stores public certificates of root CAs. It's not something related to IS. It's by default available in installed java

File: en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md (hunk 13 above)
Comment (AnuradhaSK) [reply en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md:401]: fixed in [8df37a9](https://github.com/wso2/docs-is/pull/2853/commits/8df37a9d2e6fee032cd30c4fad51df1d884d0bc3)

File: en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md (hunk 5 above)
Comment (AnuradhaSK) [reply en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md]: fixed in [8df37a9](https://github.com/wso2/docs-is/pull/2853/commits/8df37a9d2e6fee032cd30c4fad51df1d884d0bc3)

**Review by nilminiwso2 — APPROVED** [review]

File: en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md [hunk 20]
```diff
@@ -158,7 +184,7 @@ Server).
     retry= "https://localhost:8443/authenticationendpoint/retry.do"
     ```
 
-6.  Import the public certificate of the identity server to the javaca certs (or web-serverstruststore) of the JVM that the authenticationendpoint is running.
+14. Import the public certificate of the identity server to the `javaca` certs (or `web-serverstruststore`) of the JVM where the authentication endpoint is running.
```
Comment (nilminiwso2) [inline en/docs/develop/hosting-authentication-endpoint-on-a-different-server.md:187]:
```suggestion
14. Import the public certificate of the identity server to the `java cacerts` (or `web-serverstruststore`) of the JVM where the authentication endpoint is running.
```

**Pull request merged**